integration_tests/testsuite/test_custom.py

```diff
@@ -39,19 +57,175 @@ def runTest(self):
         if not output:
             logging.debug('No custom tests specified.')
         else:
-            for test_info in json.loads(output):
+            for specification in json.loads(output):
                 test_num += 1
-                name = test_info.get('name', 'test_{0}'.format(test_num))
-                path = test_info.get('path')
-                if path is None:
-                    logging.warn('Test \'%s\' has no path specified! '
-                                 'Skipping...', name)
-                    continue
+                self._run_test_for_specification(specification, test_num)
+
+    def _run_test_for_specification(self, specification, test_num):
+        """Given the specification for a test execute the steps and
+            validate the result.
+
+        Args:
+            specification: Dictionary containing the specification.
+            test_num: Identifier of the test.
+
+        Returns:
+            None.
+        """
+        name = specification.get('name', 'test_{0}'.format(test_num))
+        timeout = specification.get('timeout', 500)
```

Comment:
wondering if we should make the default timeout longer than 500ms. since these are integration tests I'd default to giving them a bit longer to run, maybe something like 2-3 seconds.